packages/compass-telemetry/src/telemetry-events.ts

```diff
@@ -2678,6 +2678,14 @@ type ExperimentViewedEvent = CommonEvent<{
   payload: { test_name: string };
 }>;
 
+type CreateIndexButtonClickedEvent = CommonEvent<{
+  name: 'Create Index Button Clicked';
+  payload: {
+    flow: 'Start with Query' | 'Start with Index' | undefined;
+    context: string;
```

Comment:
Any reason we don't want this as an enum? `'Create Index Modal'` and then we can add more later when we have them.